src/charm.py

```diff
@@ -391,6 +391,11 @@ def _configure(self):  # noqa: C901
             self._stored.status["config"] = to_tuple(MaintenanceStatus("Configuring Loki"))
             return
 
+        if 0 < int(self.config["retention-period"]) < 24:
+            self._stored.status["config"] = to_tuple(
```

Comment:
We already have a `config` used elsewhere. Strictly speaking, a section should be "edited" only within one context, to avoid unexpected overwrites. For example, you could:

```suggestion
            self._stored.status["retention"] = to_tuple(
```